fylesdk/internals/network.py

```diff
@@ -0,0 +1,124 @@
+"""
+    Defining Network related operations
+"""
+
+import os
+
+import requests
+
+from . import serializers
+from .. import exceptions
+
+class Network:
+    """Class for making GET, POST requests"""
+    HOSTNAME = None
+    HEADERS = serializers.deserialize({
+            'USER_DETAILS': 'X-User-Details',
+            'AUTHORIZATION': 'Authorization',
+            'CONTENT_TYPE': 'Content-Type',
+            'USER_AGENT': 'User-Agent',
+            'FYLE_SIGNATURE': 'X-Fyle-Signature',
+            'S3_SIGNATURE': 'X-Amz-Signature'
+        })
+
+    def __init__(self):
+        self.__root_dir = os.path.split(os.getcwd())[1]
+        Network.HOSTNAME = os.environ.get('HOSTNAME', self.__root_dir)
+
+
+    def get_request(self, url, **kwargs):
+        """Create a HTTP GET request.
+
+        Parameters:
+            url (str): Url for the wanted API.
+
+        Returns:
+            A response from the request (dict).
+        """
+
+        return self._get(url, **kwargs)
+
+
+    def post_request(self, url, **kwargs):
+        """Create a HTTP post request.
+
+        Parameters:
+            url (str): Url for the wanted API.
+
+        Returns:
+            A response from the request (dict).
+        """
+        return self._post(url, **kwargs)
+
+
+    def delete_request(self, url, **kwargs):
+        """Create a HTTP delete request.
+
+        Parameters:
+            url (str): Url for the wanted API.
+
+        Returns:
+            A response from the request (dict).
+        """
+        return self._delete(url, **kwargs)
+
+
+    def _assert_response(self, response):
+        if response.status_code == 400:
```

Comment:
response should have valid json with proper error message

Reply:
updated